membership_file/migrations/0001_initial.py

```diff
@@ -0,0 +1,66 @@
+# Generated by Django 2.2.3 on 2019-07-16 13:17
+
+import datetime
+from django.conf import settings
+import django.core.validators
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Member',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('first_name', models.CharField(max_length=63)),
+                ('tussenvoegsel', models.CharField(blank=True, max_length=15, null=True)),
+                ('last_name', models.CharField(max_length=63)),
+                ('tue_card_number', models.CharField(blank=True, max_length=15, null=True, unique=True, validators=[django.core.validators.RegexValidator(message='TU/e card numbers must only consist of exactly 7 numbers. E.g. 1234567', regex='^[0-9]{7}$')])),
+                ('external_card_number', models.CharField(blank=True, max_length=15, null=True, unique=True, validators=[django.core.validators.RegexValidator(message="External card numbers must only consist of exactly 7 numbers, followed by a hyphen (-), and ended by the 'external number' which consists of exactly 3 numbers. E.g. 1234567-123", regex='^[0-9]{7}\\-[0-9]{3}$')])),
+                ('date_of_birth', models.DateField(default=datetime.date(1970, 1, 1))),
+                ('email', models.EmailField(max_length=254, unique=True)),
+                ('phone_number', models.CharField(blank=True, max_length=16, null=True, unique=True, validators=[django.core.validators.RegexValidator(message="Phone number must be entered in the format: '+999999999'. Up to 14 digits allowed.", regex='^\\+[0-9]{8,15}$')])),
+                ('member_since', models.DateField(default=datetime.date.today)),
+                ('last_updated_date', models.DateTimeField(auto_now=True)),
+                ('user', models.OneToOneField(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='related_user', to=settings.AUTH_USER_MODEL)),
```

Comment:
Members (in the membership file) are not immediately linked to users. Users can create accounts of their own (username, email, and password), but board members will have to link a user to a member. This way, I cannot create an account to access your data; a board member will have to link the user to the member!
Moreover, it allows users to already create username-password combinations without the board needing those.
Furthermore, not every member of the knights may wish to have an account. With this setup these members will still be present in the membership file without them needing an account.

Reply:
I think there will be a lot of technical overhead to deal with this in other parts of the application and I think it is nicer to store them in the same table: we can just create accounts for all members and have them supply a password via the password reset function. Users who don't want to use the app don't reset their password and users who aren't members of the Knights don't exist in our database. Some parts of the membership file will be required in other parts of the app and having to deal with a 1-1 relation will become annoying.

Reply:
Changing the default user-model in Django is way more of a hassle than enforcing a 1-to-1 relation. Moreover, there are quite a few ways to make dealing with this not annoying at all (Seralizers play a big role in this, QuerySets allow for reverse-accessing, and Models handle accessing automatically).

Note that it is still possible to prevent everyone from creation accounts, but still allow board members to create accounts for others (so that only members will have accounts within the application).